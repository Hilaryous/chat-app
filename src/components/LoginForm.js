// @flow
import React, { PureComponent } from 'react'
import styled from 'react-emotion'
import AppContext from '../context'
import Button from './Button'
import Input from './Input'

type State = {
  email: string,
  password: string,
}

type Props = {
  signInUser: () => void,
}

class LoginForm extends PureComponent<Props, State> {
  initialState = {
    email: '',
    password: '',
  }

  state = this.initialState

  handleClearForm = () => {
    this.setState(() => this.initialState)
  };

  handleUpdateEmail= (e) => {
    this.setState({ email: e.target.value })
  }

  handleUpdatePassword= (e) => {
    this.setState({ password: e.target.value })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const { email, password } = this.state
    const { signInUser } = this.props
    signInUser(email, password)
  }

  render() {
    const { email, password } = this.state
    return (
      <Form onSubmit={this.handleSubmit}>
        <Input
          name="email"
          placeholder="Email"
          onChange={this.handleUpdateEmail}
          value={email}
        />
        <Input
          autoComplete="current-password"
          name="password"
          placeholder="Password"
          type="password"
          onChange={this.handleUpdatePassword}
          value={password}
        />
        <Button type="submit">
            Login
        </Button>
      </Form>
    )
  }
}

export default () => (
  <AppContext.Consumer>
    {({ actions: { signInUser } }) => <LoginForm signInUser={signInUser} />}
  </AppContext.Consumer>
)

//------------------------------

const Form = styled('form')({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  width: '350px',
})
