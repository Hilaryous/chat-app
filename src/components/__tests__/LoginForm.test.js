import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import LoginForm from '../LoginForm'
import AppContext from '../../context'

describe('LoginForm', () => {
  it('shows the correct snapshot tree for the LoginForm component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>

        <AppContext.Provider>
          <LoginForm />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
