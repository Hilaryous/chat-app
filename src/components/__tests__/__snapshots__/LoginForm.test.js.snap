// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm shows the correct snapshot tree for the LoginForm component 1`] = `
<form
  className="css-15dx3ku"
  onSubmit={[Function]}
>
  <input
    className="css-zof107"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    value=""
  />
  <input
    autoComplete="current-password"
    className="css-zof107"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    className="css-bjolns"
    type="submit"
  >
    Login
  </button>
</form>
`;
